src/nl/hannahsten/texifyidea/run/latex/ui/LatexSettingsEditor.kt

```diff
@@ -378,10 +380,50 @@ class LatexSettingsEditor(private var project: Project?) : SettingsEditor<LatexR
         panel.add(compilerPath)
     }
 
+    private fun updatePdfViewerComboBox() {
+        val viewers = InternalPdfViewer.availableSubset().filter { it != InternalPdfViewer.NONE } +
+            ExternalPdfViewers.getExternalPdfViewers() +
+            listOf(InternalPdfViewer.NONE)
+
+        pdfViewer.component.removeAllItems()
+        for (i in viewers.indices) {
+            (pdfViewer.component as ComboBox<PdfViewer>).addItem(viewers[i])
+        }
+        pdfViewer.updateUI()
+    }
+
     /**
      * Optional custom path for SumatraPDF.
      */
     private fun addSumatraPathField(panel: JPanel) {
+        class PathInputVerifier : InputVerifier() {
+
+            @Deprecated("Deprecated in Java")
+            override fun shouldYieldFocus(input: JComponent?): Boolean {
+                if (!verify(input)) {
+                    DialogBuilder().apply {
+                        setTitle("SumatraPDF Custom Path Invalid")
+                        setCenterPanel(
+                            JLabel(
+                                "<html>Custom Path given in run configuration of SumatraPDF doesn't contain SumatraPDF.exe. Input a valid path or leave it empty.</html>",
+                                AllIcons.General.WarningDialog,
+                                SwingConstants.LEADING
+                            )
+                        )
+                        show()
+                    }
+                }
+
+                updatePdfViewerComboBox()
+
+                return true
+            }
+
+            override fun verify(input: JComponent?): Boolean {
```

Comment:
Verification of the run configuration should happen in `LatexRunConfiguration#checkConfiguration`. Then I think it's better to leave out focus changes.
Updating the pdf viewer combobox can be done from here, I think we do similar things when selecting compilers

Reply:
Alright, I removed yieldFocusChange. I'm using the InputVerifier to update the PDFViewer ComboBox, so that the user knows if the custom path leaded to a correct directory. The user can select Sumatra as PDFViewer right after clicking away from the TextBox. Let me know if that's okay.

Reply:
Yes, verifying input is good, so I'm a bit confused why you removed the verification from here but did not add it to `checkConfiguration`? We should at least let the user know if the path is wrong, using either of the two ways.
Just for consistency, I suggested to use `checkConfiguration`, though your original solution also works of course.

Reply:
You're right. I didn't realize when ```LatexRunConfiguration#checkConfiguration``` was called, that's why I didn't use it. I also removed the verification because I wasn't sure how to do it without ```yieldFocusChange```. I shouldn't have committed right away, because I found a way. :) I committed a new version that lets the user know if the path is wrong using ```checkConfiguration``` and still uses the ```InputVerifier``` to update the PDFViewer ComboBox. I don't know if there is a better way to do it and to still update the ComboBox when ```enableSumatraPath``` is unchecked. Let me know if there's a better way or a modification to be made. :)

Reply:
We do something similar in `LatexSettingsEditor#resetEditorFrom` to show/hide certain elements depending on the compiler chosen, but I like your solution better for this case. I'll double check tomorrow, but if it works I think the PR is all good!